components/GristTableViewer/GristTableViewer.vue

```diff
@@ -0,0 +1,276 @@
+
+const total = computed(() => {
+  if (!props.totalColumn || !data.value.length) {
+    return null
+  }
+  
+  const sum = data.value.reduce((acc, record) => {
+    const value = record.fields[props.totalColumn!]
+    const num = Number(value)
+    return acc + (isNaN(num) ? 0 : num)
+  }, 0)
+  
+  return sum
+})
+
+function formatNumber(value: any): string {
+  if (value === null || value === undefined) {
+    return '-'
+  }
+  
+  const num = Number(value)
+  if (isNaN(num)) {
+    return String(value)
+  }
+  
+  return num.toLocaleString('fr-FR')
+}
+
+function getFieldValue(record: GristRecord, column: string): string {
+  const value = record.fields[column]
+  if (value === null || value === undefined) {
+    return '-'
+  }
+  return formatNumber(value)
+}
+
+async function fetchData() {
+  if (!props.url) {
+    error.value = 'URL non définie'
+    loading.value = false
+    return
+  }
+  
+  loading.value = true
+  error.value = null
+  
+  try {
+    const response = await fetch(props.url)
+    
+    if (!response.ok) {
+      throw new Error(`Erreur HTTP: ${response.status} ${response.statusText}`)
+    }
+    
+    const result: GristResponse = await response.json()
+    
+    if (!result.records) {
+      throw new Error('Format de réponse invalide: propriété "records" manquante')
+    }
+    
+    data.value = result.records
+  } catch (err) {
+    console.error('Erreur lors de la récupération des données Grist:', err)
+    error.value = err instanceof Error ? err.message : 'Erreur inconnue'
+  } finally {
+    loading.value = false
+  }
+}
+
+watch(() => props.url, () => {
+  if (props.url) {
+    fetchData()
+  }
+})
+
+onMounted(() => {
+  if (props.url) {
+    fetchData()
+  }
+})
+</script>
+
+<style scoped>
+.grist-table-viewer {
+  width: 100%;
+}
+
+.fr-table {
+  --lh: 1.125rem;
+  --max-lines: 2;
+}
+
+.fr-table table {
+  table-layout: auto !important;
+}
+
+.fr-table table :deep(th) {
+  border-right: none;
+  border-top: none;
+  border-bottom: 3px solid var(--blue-france-500);
+  border-left: none;
+  padding: 0.34375rem !important;
+  text-transform: uppercase;
+  font-size: 1.05rem;
+  line-height: var(--lh);
+  text-align: left;
+  height: 2.8rem;
+}
+
+.fr-table table :deep(tbody tr:nth-child(even)) {
+  background-color: var(--background-alt-grey);
+  --hover: var(--background-alt-grey-hover);
+  --active: var(--background-alt-grey-active);
+}
+
+.fr-table table :deep(td) {
+  padding: 0.8rem 0.3rem !important;
+  border-right: none;
+  border-left: none;
+  border-top: none;
+  border-bottom: 1px solid #cecece;
+  height: calc(var(--lh) * var(--max-lines) + 1rem);
+  font-size: 1.05rem;
+}
+
+.fr-table table :deep(*) {
+  font-size: 1.05rem;
+  line-height: var(--lh);
+}
+
+.fr-table :deep(tbody), .fr-table :deep(thead) {
+  background-color: var(--background-default-grey);
+  --hover: var(--background-default-gray-hover);
+  --active: var(--background-default-gray-active);
+}
+
+.style-cell {
```

Comment:
Do we really want to limit the height here ?
If we want to go with a defined height, we can change it to `calc(3*var(--lh))` to avoid the cut at the bottom of the text